api/src/test/java/org/pmiops/workbench/db/model/StorageEnumsTest.java

```diff
@@ -1,42 +1,117 @@
 package org.pmiops.workbench.db.model;
 
 import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
 
 import com.google.common.collect.BiMap;
-import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.junit.Test;
+import org.pmiops.workbench.model.Domain;
 
 public class StorageEnumsTest {
-  @Test
-  public void noMissingMapEntries() throws Exception {
-    for (Field f : DbStorageEnums.class.getDeclaredFields()) {
-      if (f.getType() != BiMap.class) {
-        continue;
-      }
+  private final Object INDICATES_STATIC_METHOD = null;
 
-      Class enumClass =
-          (Class) ((ParameterizedType) f.getAnnotatedType().getType()).getActualTypeArguments()[0];
+  final Set<Class> enumClasses = getEnumerationClasses();
+  final Set<Method> methods =
+      Stream.of(
+              DbStorageEnums.class.getDeclaredMethods(),
+              CommonStorageEnums.class.getDeclaredMethods())
+          .flatMap(Stream::of)
+          .collect(Collectors.toSet());
 
-      Method enumToShort = null;
-      Method shortToEnum = null;
-      for (Method m : DbStorageEnums.class.getDeclaredMethods()) {
-        if (m.getParameterTypes()[0].equals(enumClass)) {
-          enumToShort = m;
-        }
+  // e.g. public static Short reviewStatusToStorage(ReviewStatus s)
+  final Map<Class, Method> enumClassToStorageMethod =
+      methods.stream()
+          // domainToDomainId is stringly typed - test with test_domainId
+          .filter(m -> !m.getName().equals("domainToDomainId"))
+          .filter(m -> enumClasses.contains(firstMethodParameterType(m)))
+          .collect(Collectors.toMap(this::firstMethodParameterType, m -> m));
 
-        if (m.getReturnType().equals(enumClass)) {
-          shortToEnum = m;
-        }
-      }
+  // e.g. public static ReviewStatus reviewStatusFromStorage(Short s)
+  final Map<Type, Method> storageMethodToEnumClass =
+      methods.stream()
+          // domainIdToDomain is stringly typed - test with test_domainId
+          .filter(m -> !m.getName().equals("domainIdToDomain"))
+          .filter(m -> enumClasses.contains(m.getReturnType()))
+          .collect(Collectors.toMap(Method::getReturnType, m -> m));
 
-      for (Object e : enumClass.getEnumConstants()) {
-        Short shortVal = (Short) enumToShort.invoke(null, e);
+  // special-case test for domain ID round trip
+
+  @Test
+  public void test_domainId() {
+    for (Domain v : Domain.values()) {
+      String storageValue = CommonStorageEnums.domainToDomainId(v);
+      assertWithMessage("unmapped enum value: " + v).that(storageValue).isNotNull();
+      assertThat(v).isEqualTo(CommonStorageEnums.domainIdToDomain(storageValue));
```

Comment:
nit: I'd put `v` on the right, since it's the known quantity and not the experimental quantity.

Reply:
ok